cellprofiler/utilities/version.py

```diff
@@ -92,10 +92,23 @@ def get_version():
         import cellprofiler.frozen_version
         return cellprofiler.frozen_version.version_string
 
+def get_dotted_version():
```

Comment:
My workflow often is:

* Receive bug report with git hash
* Check out CellProfiler detached head at that git hash
* Confirm the bug at the git hash
  * if a bug, then check out master
    * if not a bug at master, tell the user that it's fixed
    * if a bug, fix

Mostly, people were using the trunk build as the distribution channel, no tagging, so I needed the git hash. We're releasing frequently enough that almost always, a bug report with a version # will be as good as a git hash.

The other cool thing about the SHA-1 has to do with reproducible research. If someone puts a CellProfiler pipeline in their paper or distributes the experiment measurements, the SHA-1 names the version and that could be a fork or a branch of the repo. To me, it has more of a real flavor: the thing itself rather than the name we gave to the thing. And, in BatchProfiler, we do stuff like pull out the git hash from the pipeline and run CellProfiler on that version.